tds-itemselection-impl/src/main/java/tds/itemselection/msb/MsbAssessmentSelectionService.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Educational Online Test Delivery System
+ * Copyright (c) 2016 Regents of the University of California
+ *
+ * Distributed under the AIR Open Source License, Version 1.0
+ * See accompanying file AIR-License-1_0.txt or at
+ * http://www.smarterapp.org/documents/American_Institutes_for_Research_Open_Source_Software_License.pdf
+ *
+ * SmarterApp Open Source Assessment Software Project: http://smarterapp.org
+ * Developed by Fairway Technologies, Inc. (http://fairwaytech.com)
+ * for the Smarter Balanced Assessment Consortium (http://smarterbalanced.org)
+ ******************************************************************************/
+
+package tds.itemselection.msb;
+
+import AIR.Common.DB.SQLConnection;
+import tds.itemselection.base.ItemCandidatesData;
+import tds.itemselection.base.ItemGroup;
+import tds.itemselection.loader.SegmentCollection2;
+import tds.itemselection.loader.TestSegment;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * This class handles the logic behind the selection of Multi-Stage Braille (MSB) assessments. MSB assessments
+ * (currently) consist of an adaptive segment followed by a fixed form segment. The fixed segment is not known at
+ * the start of the assessment - it must be calculated based on the student's performance in the initial adaptive
+ * section. This class accomplishes this selection by creating an artificial segment made up of item groups representing
+ * the entire pool of each of the segments. It passes these item groups into the existing adaptive selection algorithm
+ * and uses the result to determine which fixed form test is appropriate for the MSB student being assessed.
+ *
+ * @see TestSegment
+ * @see ItemGroup
+ * @see ItemCandidatesData
+ */
+public interface MsbAssessmentSelectionService {
+
+    /**
+     * Convenience method for returning the selected item candidate using the existing AdaptiveSelector2013 code. Calls
```

Comment:
Like the comments here, but could you add some quick ones to the parameters as well.